src/cpp/client/channel_cc.cc

```diff
@@ -116,10 +117,11 @@ internal::Call Channel::CreateCall(const internal::RpcMethod& method,
     } else if (!host_.empty()) {
       host_str = host_.c_str();
     }
-    grpc_slice method_slice = SliceFromCopiedString(method.name());
+    grpc_slice method_slice =
```

Comment:
@vjpai It seems to me `method.name()` should be a `const char*` internally. I think we can just use `grpc_slice_from_static_string` here. Do you think it is okay?

Reply:
It's an interesting topic. It would be a static string for anything coming from codegen, but there's nothing preventing a generic call from putting in an arbitrary crap string. So, suggestion: can we have a separate version of this function for Codegen vs generic? This is not API and it's fully internal. We'd be adding a method to an interface, but that's fine as long as the new method isn't pure (and the simplest thing to do for the new codegen-specific method in the interface would be to just have it call the general one). If you think that would be valuable, I could write that up pretty quickly, I think.

Reply:
Thank you @vjpai for the insight! I have two rather stupid questions for my own education:

1) for almost all of codegen, shouldn't will take the kRegistered branch above?
2) when we add the separate API, we still need this new SliceFromArray that I added to avoid temporary string, right?

Reply:
1. Yes, that is true. So #16719 is probably irrelevant.... I should have waited for a response before coding. Oops!
2. Yes. We're unnecessarily upconverting from a const char* to a string and then downconverting it again. This seems like a win. But rather than declare it as API in the public header, could it just be internal to this file?

Reply:
Thank you very much Vijay!
1. Thank you for the confirmation and the PR. I learned a lot of internals by just looking at the PR.
2. Yes, great point, will move here as an static function for the next commit.

Reply:
Thank you for the explanation @vjpai!

I moved the functions to the anonymous namespace in channel_cc.cc.

Reply:
@vjpai I agree we should not use static_buffer here from the current code, though I think this could be a potential problem because RpcMethod takes in a `const char *` at construction time yet the underlying string may not have a matching lifetime.